### PR TITLE
[DROOLS-6420] property reactivity reacts to all rules when a public f…

### PR DESCRIPTION
**Pull request by tkobayas**
…ield is modified in exec-model

**JIRA**: 

https://issues.redhat.com/browse/DROOLS-6420

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

File: drools-model/drools-model-compiler/src/test/java/org/drools/modelcompiler/PropertyReactivityTest.java [hunk 1]
```diff
@@ -1406,4 +1406,43 @@ public void testUpdateNonPropertyInMvel() {
...
+        PublicPerson p = new PublicPerson("John", 40);
+        ksession.insert(p);
+        int fired = ksession.fireAllRules(10);
+
+        assertEquals(3, fired);
+        assertEquals(41, p.age);
+    }
+
+    public static class PublicPerson {
+
```
Comment (pibizza) [inline drools-model/drools-model-compiler/src/test/java/org/drools/modelcompiler/PropertyReactivityTest.java:1439]: Why we need a new class for this test?

File: drools-model/drools-model-compiler/src/test/java/org/drools/modelcompiler/PropertyReactivityTest.java [hunk 2]
```diff
@@ -1406,4 +1406,43 @@ public void testUpdateNonPropertyInMvel() {
...
+                           "    $p : PublicPerson(age == 41)\n" +
+                           "then\n" +
+                           "end\n";
+
+        KieSession ksession = getKieSession(str);
+
+        PublicPerson p = new PublicPerson("John", 40);
+        ksession.insert(p);
+        int fired = ksession.fireAllRules(10);
+
```
Comment (pibizza) [inline drools-model/drools-model-compiler/src/test/java/org/drools/modelcompiler/PropertyReactivityTest.java:1434]: This 10 does not seem required.

Comment (tkobayas) [reply drools-model/drools-model-compiler/src/test/java/org/drools/modelcompiler/PropertyReactivityTest.java:1434]: @pibizza Without the fix, this rule causes an infinite loop so this is a safe guard. We can also stop such a loop by `@Test(timeout = 5000)` but setting a limit for fireAllRules is efficient.

**Comment by tkobayas** [comment]
Jenkins retest this

**Review by hellowdan — APPROVED** [review]

**Review by mariofusco — APPROVED** [review]

**Pull request merged**